packages/agreement-process/test/activateAgreement.test.ts

```diff
@@ -1500,6 +1811,33 @@ describe("activate agreement", () => {
       ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
     });
 
+    it("should throw an operationNotAllowed error when the requester is the Producer but it is not the delegate", async () => {
```

Comment:
Here but also in all the other new tests added, could you be more explicit about the fact that we talk about  "producer delegation" / "producer delegate", and say that when the producer cannot do something it's because there is an active producer delegation

```suggestion
    it("should throw an operationNotAllowed error when the requester is the Producer, there is an active producer delegation, but requester is not the delegate", async () => {
```

Reply:
We will resolve that after develop merge, [PIN](https://pagopa.atlassian.net/browse/PIN-5773)